File: front-service/src/app/models/meal.ts
import {MealFood} from "./mealfood";

export interface Meal {
  id: number;
  mealType: string; // Тип приема пищи: завтрак, обед или ужин
  dayId: number;    // ID дня, к которому привязан прием пищи
  mealFoods: MealFood[]; // Продукты, связанные с приемом пищи
}

File: meal-service/src/main/java/org/ilya/entities/Meal.java
package org.ilya.entities;

import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.List;
import java.time.LocalDate;

@Data
@NoArgsConstructor
@Entity
@Table(name = "meals")
public class Meal {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "meal_type", nullable = false)
    private String mealType;

    @ManyToOne
    @JoinColumn(name = "day_id", nullable = false)
    private Day day;

    @OneToMany(mappedBy = "meal", cascade = CascadeType.ALL, fetch = FetchType.EAGER)
    private List<MealFood> mealFoods;

}
